spectrocloud/resource_cluster_libvirt.go

```diff
@@ -0,0 +1,643 @@
+			},
+			"os_patch_after": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateDiagFunc: validateOsPatchOnDemandAfter,
+			},
+			"kubeconfig": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloud_config": {
+				Type:     schema.TypeList,
+				ForceNew: true,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ssh_key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"vip": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"pack": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"tag": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"values": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"machine_pool": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Set:      resourceMachinePoolLibvirtHash,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"control_plane": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+							//ForceNew: true,
+						},
+						"control_plane_as_worker": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+
+							//ForceNew: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							//ForceNew: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"update_strategy": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "RollingUpdateScaleOut",
+						},
+						"instance_type": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"attached_disks_size_gb": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"cpus_sets": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"disk_size_gb": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"memory_mb": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"cpu": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"placements": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"appliance_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"network_type": {
+										Type:     schema.TypeString,
+										Required: true,
```

Comment:
We can add a constraint to allow just `[default, bridge]` in `network_type`

Reply:
@rishi-anand added it.